pkg/controller/perconaservermongodb/pbm_test.go

```diff
@@ -0,0 +1,261 @@
+			&config.Config{
+				Storage: config.StorageConf{
+					Type: storage.S3,
+					S3: &s3.Config{
+						Bucket:      "operator-testing",
+						Region:      "us-east-1",
+						EndpointURL: "https://s3.amazonaws.com",
+					},
+				},
+			},
+			&config.Config{
+				Storage: config.StorageConf{
+					Type: storage.S3,
+					S3: &s3.Config{
+						Bucket:         "operator-testing",
+						Region:         "us-east-1",
+						EndpointURL:    "https://s3.amazonaws.com",
+						MaxUploadParts: 2000,
+					},
+				},
+			},
+			false,
+		},
+		{
+			"azure: endpointUrl changed",
+			&config.Config{
+				Storage: config.StorageConf{
+					Type: storage.Azure,
+					Azure: &azure.Config{
+						Account:     "operator-account",
+						Container:   "operator-testing",
+						EndpointURL: "https://accountName.blob.core.windows.net",
+					},
+				},
+			},
+			&config.Config{
+				Storage: config.StorageConf{
+					Type: storage.Azure,
+					Azure: &azure.Config{
+						Account:     "operator-account",
+						Container:   "operator-testing",
+						EndpointURL: "https://accountName-1.blob.core.windows.net",
+					},
+				},
+			},
+			true,
+		},
+		{
+			"azure: container changed",
+			&config.Config{
+				Storage: config.StorageConf{
+					Type: storage.Azure,
+					Azure: &azure.Config{
+						Account:     "operator-account",
+						Container:   "operator-testing",
+						EndpointURL: "https://accountName.blob.core.windows.net",
+					},
+				},
+			},
+			&config.Config{
+				Storage: config.StorageConf{
+					Type: storage.Azure,
+					Azure: &azure.Config{
+						Account:     "operator-account",
+						Container:   "operator-testing-1",
+						EndpointURL: "https://accountName.blob.core.windows.net",
+					},
+				},
+			},
+			true,
+		},
+		{
+			"azure: account changed",
+			&config.Config{
+				Storage: config.StorageConf{
+					Type: storage.Azure,
+					Azure: &azure.Config{
+						Account:     "operator-account",
+						Container:   "operator-testing",
+						EndpointURL: "https://accountName.blob.core.windows.net",
+					},
+				},
+			},
+			&config.Config{
+				Storage: config.StorageConf{
+					Type: storage.Azure,
+					Azure: &azure.Config{
+						Account:     "operator-account-1",
+						Container:   "operator-testing",
+						EndpointURL: "https://accountName.blob.core.windows.net",
+					},
+				},
+			},
+			true,
+		},
+		{
+			"fs: path changed",
+			&config.Config{
+				Storage: config.StorageConf{
+					Type: storage.Filesystem,
+					Filesystem: &fs.Config{
+						Path: "/mnt/backups",
+					},
+				},
+			},
+			&config.Config{
+				Storage: config.StorageConf{
+					Type: storage.Filesystem,
+					Filesystem: &fs.Config{
+						Path: "/mnt/backups-1",
+					},
+				},
+			},
+			true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := isResyncNeeded(tt.currentCfg, tt.newCfg); got != tt.expected {
+				t.Errorf("%s: got %v, want %v", tt.name, got, tt.expected)
+			}
```

Comment:
Optional: since we have access to the `testify assert` in this operator, we can use it without problem

```
		t.Run(tt.name, func(t *testing.T) {
			assert.Equal(t, tt.expected, isResyncNeeded(tt.currentCfg, tt.newCfg))
		})
```